### PR TITLE
Add context package

### LINKED ISSUE #189: Use context
Using `context.Context` could improve the reability and flexibility of the `gopass` source code in several ways.

On the one hand it could used communicate per-invocation variables, like `isTerminal` but also allow flexible operation timeouts.

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #189

### CODE REVIEW

**Comment by dominikschulz** [comment]
@metalmatze @zwopir This should be merged ASAP, rebasing this branch would be a huge pain.

File: utils/ctxutil/ctxutil.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package ctxutil
+
```
Comment (zwopir) [inline utils/ctxutil/ctxutil.go]: code-wise this package is fine. Personally I prefer putting the 'ctx getter/setter' funcs in the package that defines the value that is transported via context.
This way you're mixing the tooling (ctx/ctxutils) with the business logic (default values). So another refactoring steps may be 
 - moving the getter/setter funcs
 - exposing the type assertion results `ok` (`value, ok := ctx.Value(key).(type)` as getter func return value.

File: utils/ctxutil/ctxutil_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		t.Errorf("Should have debug flag")
+	}
+	if IsDebug(WithDebug(context.Background(), false)) {
+		t.Errorf("Should not have debug flag")
+	}
+}
+
+func TestColor(t *testing.T) {
+	if !IsColor(context.Background()) {
+		t.Errorf("Vanialla ctx should have color")
```
Comment (zwopir) [inline utils/ctxutil/ctxutil_test.go]: typo

File: action/clipboard_others.go [hunk 3]
```diff
@@ -15,10 +16,10 @@ import (
 // process group until the timeout is expired. It will then compare the contents
 // of the clipboard and erase it if it still contains the data gopass copied
 // to it.
-func clearClipboard(content []byte, timeout int) error {
+func clearClipboard(ctx context.Context, content []byte, timeout int) error {
 	hash := fmt.Sprintf("%x", sha256.Sum256(content))
 
-	cmd := exec.Command(os.Args[0], "unclip", "--timeout", strconv.Itoa(timeout))
+	cmd := exec.CommandContext(ctx, os.Args[0], "unclip", "--timeout", strconv.Itoa(timeout))
```
[the remark this answers is not in the document]
Comment (dominikschulz) [reply action/clipboard_others.go]: Good idea, will do

File: action/edit.go [hunk 4]
```diff
@@ -87,10 +88,12 @@ func (s *Action) editor(content []byte) ([]byte, error) {
 
 	editor = cmdArgs[0]
 	args := append(cmdArgs[1:], tmpfile.Name())
-	cmd := exec.Command(editor, args...)
+
+	cmd := exec.CommandContext(ctx, editor, args...)
```
[the remark this answers is not in the document]
Comment (dominikschulz) [reply action/edit.go]: Yes, getting the editor `kill -9`ed is not nice. Switching to `exec.Command`.

File: action/insert.go [hunk 5]
```diff
@@ -52,60 +54,60 @@ func (s *Action) Insert(c *cli.Context) error {
...
 		return nil
 	}
 
 	if !force { // don't check if it's force anyway
-		if s.Store.Exists(name) && !s.askForConfirmation(fmt.Sprintf("An entry already exists for %s. Overwrite it?", name)) {
-			return s.exitError(ExitAborted, nil, "not overwriting your current secret")
+		if s.Store.Exists(name) && !s.askForConfirmation(ctx, fmt.Sprintf("An entry already exists for %s. Overwrite it?", name)) {
+			return s.exitError(ctx, ExitAborted, nil, "not overwriting your current secret")
 		}
 	}
```
[the remark this answers is not in the document]
Comment (dominikschulz) [reply action/insert.go]: Yes, definitely. There are a few more candidates for context values as well. This PR is a mere first step at contextualization.

File: store/sub/fsck.go [hunk 6]
```diff
@@ -50,14 +51,14 @@ func (s *Store) mkStoreWalkerFsckFunc(prefix string, check, force bool, storeRec
...
+			return s.fsckCheckDir(ctx, prefix, path, check, force, askFunc, shadowMap, countFn)
 		}
-		return s.fsckCheckFile(prefix, path, check, force, storeRec, askFunc, shadowMap, countFn)
+		return s.fsckCheckFile(ctx, prefix, path, check, force, storeRec, askFunc, shadowMap, countFn)
 	}
 }
 
 // fsckCheckDir checks a directory, mostly for it's permissions
-func (s *Store) fsckCheckDir(prefix, fn string, check, force bool, askFunc func(string) bool, sh map[string]struct{}, countFn func(string)) error {
+func (s *Store) fsckCheckDir(ctx context.Context, prefix, fn string, check, force bool, askFunc func(context.Context, string) bool, sh map[string]struct{}, countFn func(string)) error {
```
[the remark this answers is not in the document]
Comment (dominikschulz) [reply store/sub/fsck.go]: Yes, will change in a later PR.

File: utils/ctxutil/ctxutil.go (hunk 1 above)
Comment (dominikschulz) [reply utils/ctxutil/ctxutil.go]: I didn't want to expose the type assertion results, as this makes using these method a little more annyoing, but you're right about mixing tooling with business logic. Will change that in a later PR. Right now the `ctxutil` is IMHO OK as it helped a lot in quickly introducing contexts.

File: utils/ctxutil/ctxutil_test.go (hunk 2 above)
Comment (dominikschulz) [reply utils/ctxutil/ctxutil_test.go]: Haha, thanks! That was intentional to check if you read everything :smirk: (not really)

**Pull request merged**